src/runtime_src/core/pcie/tools/xbutil/xbutil.h

```diff
@@ -148,6 +148,53 @@ static const std::map<MEM_TYPE, std::string> memtype_map = {
 
 static const std::map<std::string, command> commandTable(map_pairs, map_pairs + sizeof(map_pairs) / sizeof(map_pairs[0]));
 
+class xclbin {
```

Comment:
This class looks like a mix of responsibilities.  It is better to separate the device concept from the xclbin concept.  Have  a device class that opens the device and manages this.  Then have this xclbin class get the uuid and open and manage a context on the xclbin itself.  

The xclbin class constructor should take a device object as argument, it looks like we already have such a class. It doesn't look like the cuidx is needed as argument since we default to virtual CU (-1).